File: common/src/main/java/com/ssblur/scriptor/word/action/teleport/GotoAction.java
package com.ssblur.scriptor.word.action.teleport;

import com.ssblur.scriptor.helpers.targetable.Targetable;
import com.ssblur.scriptor.api.word.Descriptor;

public class GotoAction extends SwapAction {
  @Override
  public void apply(Targetable caster, Targetable targetable, Descriptor[] descriptors) {
    if(targetable.getLevel().isClientSide
      || caster == null
      || caster.getLevel() == null
      || targetable.getLevel() == null) return;

    teleport(caster, targetable);
  }

  @Override
  public Cost cost() { return new Cost(6, COSTTYPE.ADDITIVE); }
}

File: common/src/main/kotlin/com/ssblur/scriptor/word/action/teleport/GotoAction.kt
package com.ssblur.scriptor.word.action.teleport

import com.ssblur.scriptor.api.word.Descriptor
import com.ssblur.scriptor.helpers.targetable.Targetable

class GotoAction : SwapAction() {
    override fun apply(caster: Targetable, targetable: Targetable, descriptors: Array<Descriptor>) {
        if (targetable.level.isClientSide) return

        teleport(caster, targetable)
    }

    override fun cost() = Cost(6.0, COSTTYPE.ADDITIVE)
}
